core/commands/storage/upload/upload/inquiry.go

```diff
@@ -0,0 +1,196 @@
+	"github.com/tron-us/go-btfs-common/utils/grpc"
+	"github.com/tron-us/protobuf/proto"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+var StorageRenewInquiryCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "inquire hosts for the new duration and return the hosts signed contracts.",
+		ShortDescription: `
+Communicate with hosts to accept or reject the new contracts, return the signed the contracts if agree with the extended period, 
+or find new hosts to download and store the shards.`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("renter-pid", true, false, "Original renter peer ID."),
+		cmds.StringArg("file-hash", true, false, "Hash of file to upload."),
+		cmds.StringArg("entry-key", true, false, "Key for querying meta data from Guard."),
+	},
+	RunTimeout: 5 * time.Minute,
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		n, err := cmdenv.GetNode(env)
+		if err != nil {
+			return err
+		}
+		cfg, err := n.Repo.Config()
+		renterPid := req.Arguments[0]
+		fileHash := req.Arguments[1]
+		entryKey := req.Arguments[2]
+
+		metaReq := &guardpb.CheckFileStoreMetaRequest{
+			FileHash:     fileHash,
+			RenterPid:    renterPid,
+			RequesterPid: entryKey,
+			RequestTime:  time.Now().UTC(),
+		}
+		metaReq.Signature = []byte(entryKey)
+		var meta *guardpb.FileStoreStatus
+		err = grpc.GuardClient(cfg.Services.GuardDomain).WithContext(req.Context, func(ctx context.Context,
+			client guardpb.GuardServiceClient) error {
+			meta, err = client.CheckFileStoreMeta(ctx, metaReq)
+			if err != nil {
+				return err
+			}
+			return nil
+		})
+
+		var bidTotalPrice int64
+		var askTotalPrice int64
+		var askDailyPrice int64
+		contracts := meta.Contracts
+		hostPidMap := make(map[string]string)
+		contractList := make([]*guardpb.Contract, 0)
+		storageLength := (int)(contracts[0].RentEnd.Sub(contracts[0].RentStart).Hours() / 24)
+		for _, contract := range contracts {
+			if contract.State == guardpb.Contract_UPLOADED {
+				hostPidMap[contract.ShardHash] = contract.HostPid
+			}
+		}
+		for _, contract := range contracts {
+			if contract.State == guardpb.Contract_RECREATED {
+				hostPid := hostPidMap[contract.ShardHash]
+				ns, err := hub.GetHostSettings(req.Context, cfg.Services.HubDomain, hostPid)
+				if err != nil {
+					return err
+				}
+				askPrice := int64(ns.StoragePriceAsk)
+				askTotalPrice += uh.TotalPay(contract.ShardFileSize, askPrice, int(storageLength))
+				askDailyPrice += uh.TotalPay(contract.ShardFileSize, askPrice, 1)
+				if contract.Price != askPrice {
+					contract.Price = askPrice
+				}
+				bidPrice := contract.Amount
+				bidTotalPrice += bidPrice
+				contractList = append(contractList, contract)
+			}
+		}
+
+		signedGuardContracts := make([]*guardpb.Contract, 0)
+		if askTotalPrice <= bidTotalPrice {
+			signedGuardContracts, err = agreeProcess(req, n, renterPid, fileHash, storageLength, hostPidMap, contractList)
```

Comment:
not sure here, if original host did not agree the contract, how do you start find new host process?